### PR TITLE
[MXNET-160] Fix for issue 9062

### PR DESCRIPTION
**Pull request by haojin2**
## Description ##
Bug fix for issue #9062 

### Essentials ###
- [x] The PR title starts with [MXNET-160]
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Fix the typo in Shape function of sample_multinomial_op.h
- [x] Add test coverage to verify the fix

### CODE REVIEW

**Comment by haojin2** [comment]
@eric-haibin-lin @rahul003 @reminisce @anirudh2290 This is bug fix for #9062, please give a review if you have a minute. Thanks!

File: tests/python/unittest/test_random.py [hunk 1]
```diff
@@ -276,29 +276,34 @@ def test_parallel_random_seed_setting():
...
+        if len(x.shape) is 1:
+            x = x.reshape((1, x.shape[0]))
+            dx = dx.reshape(1, dx.shape[0])
+            y = y.reshape((1, y.shape[0]))
+            prob = prob.reshape((1, prob.shape[0]))
+        for i in range(x.shape[0]):
+            freq = np.bincount(y[i,:], minlength=5)/np.float32(samples)*x[i,:].sum()
+            mx.test_utils.assert_almost_equal(freq, x[i], rtol=0.20)
+            rprob = x[i][y[i]]/x[i].sum()
+            mx.test_utils.assert_almost_equal(np.log(rprob), prob.asnumpy()[i])
```
Comment (anirudh2290) [inline tests/python/unittest/test_random.py]: should we provide specific atol ?

File: tests/python/unittest/test_random.py [hunk 2]
```diff
@@ -276,29 +276,34 @@ def test_parallel_random_seed_setting():
...
+
+            real_dx = np.zeros((5,))
+            for j in range(samples):
+                real_dx[y[i][j]] += 5.0 / rprob[j]
+            mx.test_utils.assert_almost_equal(real_dx, dx[i, :], rtol=1e-4)
```
Comment (anirudh2290) [inline tests/python/unittest/test_random.py]: same question as above.

File: tests/python/unittest/test_random.py (hunk 1 above)
Comment (haojin2) [reply tests/python/unittest/test_random.py]: Is the default not 1e-20?

Comment (haojin2) [reply tests/python/unittest/test_random.py]: The current test can always pass, I guess the default 1e-20 value should be okay?

Comment (reminisce) [reply tests/python/unittest/test_random.py]: Should provide `atol` to avoid failure when comparing two numbers that are close to zero.

Comment (haojin2) [reply tests/python/unittest/test_random.py]: Okay, thanks!

**Pull request merged**